// MODULE: m1-common
// FILE: common.kt

<!CONFLICTING_OVERLOADS, NO_ACTUAL_FOR_EXPECT!>expect fun foo()<!>
<!CONFLICTING_OVERLOADS, NO_ACTUAL_FOR_EXPECT!>expect fun foo()<!>

<!NO_ACTUAL_FOR_EXPECT!>expect fun foo(x: Int)<!>
